crm_compassion/model/demand_planning.py

```diff
@@ -33,43 +34,84 @@ class WeeklyDemand(models.Model):
     ##########################################################################
     @api.model
     def create(self, vals):
-        vals = self.update_vals(vals)
+        self.update_vals(vals)
         res = super(WeeklyDemand, self).create(vals)
         return res
 
     @api.multi
     def write(self, vals):
-        vals = self.update_vals(vals)
+        self.update_vals(vals)
         res = super(WeeklyDemand, self).write(vals)
         return res
 
     ##########################################################################
     #                             PUBLIC METHODS                             #
     ##########################################################################
     def update_vals(self, vals):
-        starting_date = datetime.strptime(vals.get('week_start_date'),
-                                          "%Y-%m-%d")
-        if self.check_is_date_in_next_8_weeks(starting_date):
-            vals['period_locked'] = True
-        return vals
+        starting_date = vals.get('week_start_date')
+        if starting_date:
+            corrected_date = datetime.strptime(starting_date, "%Y-%m-%d")
+            if self.check_is_date_in_next_8_weeks(corrected_date):
+                vals['period_locked'] = True
+            else:
+                vals['period_locked'] = False
+        # return vals
 
     def check_is_date_in_next_8_weeks(self, date):
         return date <= (datetime.today() + timedelta(weeks=8))
 
 
 class DemandPlanning(models.Model):
-
     _name = 'demand.planning'
     _description = 'Demand Planning'
 
     name = fields.Char(required=True)
     date = fields.Date()
     sent_date = fields.Datetime()
     state = fields.Selection([
-        ('new', 'New'),
+        ('draft', 'Draft'),
         ('sent', 'Sent'),
         ('error', 'Error'),
     ])
     weekly_demand_ids = fields.Many2many(
         'demand.weekly.demand', string='Weekly Demands'
     )
+
+    ##########################################################################
+    #                             ORM METHODS                                #
+    ##########################################################################
+    @api.model
+    def create(self, vals):
+        res = super(DemandPlanning, self).create(vals)
+        res.state = 'draft'
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(DemandPlanning, self).write(vals)
+        if not isinstance(res, bool):
+            res.state = 'draft'
```

Comment:
I figured after editing a message in `sent` or `error` state, we should always put it back to `draft` until the user try to resend it again to GMC. Keeping another one of these states after editing would be incorrect.